### PR TITLE
Delete boehm specific code

### PR DESCRIPTION
**Pull request by jacob-hughes**
This is now being served from the softdev/boehm_allocator crate, which
contains code needed for both rboehm and rustc_boehm.

### CODE REVIEW

**Comment by ltratt** [comment]
A maybe-silly question: why does this repo exist? Or, maybe accurately, is its name misleading or ...? I'm a bit confused that we now have rustc_boehm, rboehm, and boehm_allocator repositories and no longer entirely sure which does which :)

**Comment by jacob-hughes** [comment]
I've pushed a README that explains the structure of the GC library a bit better. I've also renamed uses of rboehm => libgc in source.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,31 @@
...
+There are three repositories which make up the gc infrastructure:
+    * libgc: the main library which provides the `Gc<T>` smart pointer and its
+      API.
+    * libgc_internal: contains the gc allocation and collector logic. This is
+      collector specific, and can be conditionally compiled to support different
+      implementations. At the moment, it only supports a single collector
+      implementation: the Boehm-Demers-Weiser GC. Users should never interact
+      directly with this crate. Instead, any relevant APIs are re-exported
+      through libgc.
+    * rustgc: a fork of rustc with GC-aware optimisations. This can be used to
```
Comment (ltratt) [inline README.md]: Should we say something like "Use of rustgc is not mandated, but it enables further optimisations for programs which use `libgc`"?

Comment (jacob-hughes) [reply README.md]: Good idea, added here: 19fc1c185090d732dc5ab852fcdc689426bb194e (Also fixed the markdown syntax)

**Comment by ltratt** [comment]
Please squash.

**Comment by jacob-hughes** [comment]
Squashed, but we can't merge until https://github.com/softdevteam/libgc_internal/pull/1 is merged.

**Comment by jacob-hughes** [comment]
Format fixes: dd73b30a973b434b4055c2386787d37040eb309f

**Comment by ltratt** [comment]
Please squash.

**Comment by jacob-hughes** [comment]
Fixed

**Pull request merged**